cmd/util/ledger/migrations/change_contract_code_migration.go

```diff
@@ -206,64 +203,54 @@ func BurnerAddressForChain(chainID flow.ChainID) flow.Address {
 func SystemContractChanges(chainID flow.ChainID, options SystemContractChangesOptions) []SystemContractChange {
 	systemContracts := systemcontracts.SystemContractsForChain(chainID)
 
-	serviceAccountAddress := systemContracts.FlowServiceAccount.Address
-	idTableStakingAddress := systemContracts.IDTableStaking.Address
-	clusterQCAddress := systemContracts.ClusterQC.Address
-	dkgAddress := systemContracts.DKG.Address
-	fungibleTokenAddress := systemContracts.FungibleToken.Address
-	flowTokenAddress := systemContracts.FlowToken.Address
-	flowFeesAddress := systemContracts.FlowFees.Address
-	flowStorageFeesAddress := systemContracts.FlowStorageFees.Address
-	viewResolverAddress := systemContracts.ViewResolver.Address
-	metadataViewsAddress := systemContracts.MetadataViews.Address
-	fungibleTokenMetadataViewsAddress := common.Address(fungibleTokenAddress)
-	fungibleTokenSwitchboardAddress := common.Address(fungibleTokenAddress)
-
-	burnerAddress := BurnerAddressForChain(chainID)
-
-	var stakingCollectionAddress common.Address
-	var stakingProxyAddress common.Address
+	env := coreContractstemplates.Environment{
+		ServiceAccountAddress:             systemContracts.FlowServiceAccount.Address.Hex(),
+		ViewResolverAddress:               systemContracts.ViewResolver.Address.Hex(),
+		BurnerAddress:                     BurnerAddressForChain(chainID).Hex(),
+		FungibleTokenAddress:              systemContracts.FungibleToken.Address.Hex(),
+		NonFungibleTokenAddress:           systemContracts.NonFungibleToken.Address.Hex(),
+		MetadataViewsAddress:              systemContracts.MetadataViews.Address.Hex(),
+		FungibleTokenMetadataViewsAddress: systemContracts.FungibleToken.Address.Hex(),
+		FungibleTokenSwitchboardAddress:   systemContracts.FungibleToken.Address.Hex(),
+		FlowTokenAddress:                  systemContracts.FlowToken.Address.Hex(),
+		IDTableAddress:                    systemContracts.IDTableStaking.Address.Hex(),
+		QuorumCertificateAddress:          systemContracts.ClusterQC.Address.Hex(),
+		DkgAddress:                        systemContracts.DKG.Address.Hex(),
+		FlowFeesAddress:                   systemContracts.FlowFees.Address.Hex(),
+		StorageFeesAddress:                systemContracts.FlowStorageFees.Address.Hex(),
+	}
```

Comment:
... and here this would change to:

```suggestion
	env := systemContracts.AsTemplateEnv()
	env.BurnerAddress = BurnerAddressForChain(chainID).Hex()
```